### PR TITLE
Test and analyze constructor related cheatsheet snippets

### PR DESCRIPTION
**Pull request by parlough**
Also moves some pieces around to enable the testing of more code snippets.

Makes progress towards https://github.com/dart-lang/site-www/issues/2295

### CODE REVIEW

**Comment by parlough** [comment]
@domesticmouse Mind taking a look at these changes when you get a chance? I slightly modified some snippets and added tests for various excerpts.

File: examples/misc/lib/cheatsheet/redirecting_constructors.dart [hunk 1]
```diff
@@ -17,11 +17,11 @@ class Automobile {
...
 // #docregion const-constructors
 class ImmutablePoint {
-  const ImmutablePoint(this.x, this.y);
+  static const ImmutablePoint origin = ImmutablePoint(0, 0);
 
   final int x;
   final int y;
 
-  static const ImmutablePoint origin = ImmutablePoint(0, 0);
+  const ImmutablePoint(this.x, this.y);
```
Comment (domesticmouse) [inline examples/misc/lib/cheatsheet/redirecting_constructors.dart:25]: I'm curious why the constructor is now last? I usually put the constructors at the top of a class declaration.

**Review by domesticmouse — APPROVED** [review]

File: examples/misc/lib/cheatsheet/redirecting_constructors.dart (hunk 1 above)
Comment (parlough) [reply examples/misc/lib/cheatsheet/redirecting_constructors.dart:25]: I guess this is just the form I've seen used most commonly throughout non-Flutter Dart code and on the website, but I'm happy to switch back. Just let me know :)

Comment (domesticmouse) [reply examples/misc/lib/cheatsheet/redirecting_constructors.dart:25]: I'd suggest landing this as is, and then we can figure out the style guide issues =)

Hey @RedBrogdon, do we have a style guide somewhere on constructor placement in Dart classes?

**Comment by domesticmouse** [comment]
PTAL @kwalrath =)

**Review by kwalrath — APPROVED** [review]

**Pull request merged**